modules/gdscript/gdscript.cpp

```diff
@@ -2008,6 +2010,15 @@ void GDScriptInstance::notification(int p_notification, bool p_reversed) {
 			}
 		}
 	}
+
+	if (p_notification == Node::NOTIFICATION_DISABLED) {
+		SelfList<GDScriptFunctionState> *iterator = pending_func_states.first();
+		while (iterator != nullptr) {
+			iterator->self()->cancel();
+			iterator = iterator->next();
+		}
+	}
+	pending_func_states.clear();
```

Comment:
This shouldn't be here, it means it clears this when *any* notification is passed

Reply:
Thanks, i think you found the bug.